score-lang-cli/src/main/java/org/openscore/lang/cli/utils/CompilerHelperImpl.java

```diff
@@ -62,6 +63,16 @@ public CompilationArtifact compile(String filePath, List<String> dependencies) t
         Set<SlangSource> depsSources = new HashSet<>();
         File file = new File(filePath);
         Validate.isTrue(file.isFile(), "File: " + file.getName() + " was not found");
+        boolean validFileExtension = false;
+
+        Collection<File> slangFiles = FileUtils.listFiles(new File(file.getParent()), SLANG_FILE_EXTENSIONS, false);
+        for (File slangFile : slangFiles) {
+            if (StringUtils.equals(slangFile.getAbsolutePath(), file.getAbsolutePath())){
+                validFileExtension = true;
```

Comment:
--cp is checked afterwards.The logic is as follows. First it checks that the source file exists. Then it gets all the files from the parent directory with our specified extensions and looks for the source file within them. Afterwards it gets all the files from the classpath with those extensions